### PR TITLE
Fix to bypass CORS calls when content is not on the page current domain.  

### PR DESCRIPTION
**Pull request by sferreiro**
- Update of package.json to set specific release version for node-min…ify, latest doesn't work with that old version of node we are using.

- Updating build.js to:
	- Make it a bit more readable with nice clousures.
	- Default output folder to ../output
	- Include optional parameters: -o to be able to change output directory, -debug generates .js files without compression, -t changes the url of the translators.
	-Moved PME.js to be inside the src folder and also minified and copied into the output on build.

- Fix for CORS calls when generated iframe url doesn't match current domain (like for google scholar) by setting the document.domain property.

### CODE REVIEW

File: src/common/messaging.js [hunk 1]
```diff
@@ -52,7 +52,13 @@ Zotero.Messaging = new function() {
 	 */
 	this.receiveMessage = function(messageName, args, sendResponseCallback, tab) {
 		try {
-			//Zotero.debug("Messaging: Received message: "+messageName);
+			Zotero.debug("Messaging: Received message: "+messageName);
+
+			if (messageName == undefined) {
```
Comment (aniamy) [inline src/common/messaging.js:57]: Indentation is off here

**Review by aniamy — APPROVED** [review]

File: src/build.js [hunk 2]
```diff
@@ -1,148 +1,200 @@
...
+
+};
+
+var parseArguments = function() {
+    for(var i = 0; i < process.argv.length; i++) {
+        var arg = process.argv[i];
+        var val = process.argv[++i];
+        console.log(arg, val);
+        switch(arg) {
+            case '-o': //Output folder, default is "../"
```
Comment (m9bromu) [inline src/build.js]: Isn't the default ../output/ ?   See line 6

**Review by m9bromu — APPROVED** [review]
I just have a question about a comment

File: src/build.js (hunk 2 above)
Comment (sferreiro) [reply src/build.js]: Yes, you are correct.

**Review by ehernand — APPROVED** [review]
Nice work!

**Pull request closed without merging**